### ISSUE TITLE
Values of Rows with XLFormRowDescriptorTypeSelectorPush are not showing up in form.formValues

### ISSUE THREAD

**Issue by kpavankotesh** [xmartlabs/XLForm#1052]
Hi,

I recently updated to `4.0.1` and the rows with XLFormRowDescriptorTypeSelectorPush, valueTransformers

```swift
row = XLFormRowDescriptor(tag: "tag", rowType: XLFormRowDescriptorTypeSelectorPush, title: "title")
row.valueTransformer = CustomValueTransformer.self
```

Values of such rows are not coming up in form.formValues on submit. When I revert back to `4.0.0` it works fine. Can you please check? 

Note: I upgraded it as a part of Swift 5 migration. Please check if that could be the issue.

**Comment by chrisballinger** [comment]
I am also seeing a similar, possibly related issue where form.formValues does not return custom objects unless they conform to `XLFormOptionObject`.

See the code here: https://github.com/xmartlabs/XLForm/blob/master/XLForm/XL/Helpers/NSObject+XLFormAdditions.m#L43

**Comment by pavankotesh** [comment]
Yep @chrisballinger. Confirming the custom object to `XLFormOptionObject` works. Thanks.
